packages/dapp/hooks/useEarlyMint.js

```diff
@@ -29,9 +23,16 @@ const useMintEarly = () => {
 
         const contract = new Contract(tokenAddress, abi, signer);
 
-        const coupon = await signCouponForAddress(state.address, signer);
+        const coupons = (await import(`../data/coupons.${ethereumNetwork}.json`)).default;
+
+        const coupon = coupons.find((coupon) => {
+          const [address] = Object.keys(coupon);
+          return address === state.address;
+        });
+
+        if (!coupon) throw new Error('You are not able to mint in this state');
 
-        await contract.mintEarly(coupon, {
+        await contract.mintEarly(coupon[state.address], {
```

Comment:
Add one more check that queries the contract to see if the user has spent that coupon.